### PR TITLE
👷 Skip Safari and IE for extension Browserstack unit tests

### PR DESCRIPTION
**Pull request by cy-moi**
## Motivation
We do not need to run extension tests on Safari or IE

## Changes
Add a browser configuration for extension only karma test.

## Testing

- [x] Local
- [ ] Staging
- [ ] Unit
- [ ] End to end

---

I have gone over the [contributing](https://github.com/DataDog/browser-sdk/blob/main/CONTRIBUTING.md) documentation.

### CODE REVIEW

File: test/unit/karma.bs.conf.js [hunk 1]
```diff
@@ -1,6 +1,15 @@
-const { browserConfigurations } = require('../browsers.conf')
 const { getBuildInfos } = require('../envUtils')
-const karmaBaseConf = require('./karma.base.conf')
+const karmaBaseConfig = require('./karma.base.conf')
```
Comment (nulrich) [inline test/unit/karma.bs.conf.js:2]: **💬 suggestion:** Maybe a name like `createKarmaBaseConfig` or `initKarmaBaseConfig`.

File: test/browsers.developer-extension.conf.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
+// Capabilities: https://www.browserstack.com/automate/capabilities
+
+const extensionBrowserConfigurations = [
```
Comment (nulrich) [inline test/browsers.developer-extension.conf.js:3]: **💭 thought:** ‏Can we use the `browsers.conf.js` and filter to only get Edge FF and Chrome. So we have only one place with browsers names and versions.

File: package.json [hunk 3]
```diff
@@ -20,7 +20,7 @@
     "test": "yarn test:unit:watch",
     "test:unit": "karma start ./test/unit/karma.local.conf.js",
     "test:unit:watch": "yarn test:unit --no-single-run",
-    "test:unit:bs": "node ./scripts/test/bs-wrapper.js karma start test/unit/karma.bs.conf.js",
+    "test:unit:bs": "node ./scripts/test/bs-wrapper.js karma start test/unit/karma.bs.conf.js && karma start test/unit/karma.bs.conf.js --ext",
```
Comment (nulrich) [inline package.json:23]: **❓ question:** ‏Shouldn't we run `node ./scripts/test/bs-wrapper.js` before the second karma? Not sure BS will be ready to run a second set of tests just after the first.

File: package.json [hunk 4]
```diff
@@ -20,7 +20,7 @@
     "test": "yarn test:unit:watch",
     "test:unit": "karma start ./test/unit/karma.local.conf.js",
     "test:unit:watch": "yarn test:unit --no-single-run",
-    "test:unit:bs": "node ./scripts/test/bs-wrapper.js karma start test/unit/karma.bs.conf.js",
+    "test:unit:bs": "node ./scripts/test/bs-wrapper.js karma start test/unit/karma.bs.conf.js && node ./scripts/test/bs-wrapper.js karma start test/unit/karma.bs.conf.js --ext",
```
Comment (cy-moi) [inline package.json:23]: Question❓: shall we proceed with skipping the Browserstack tests for the extension completely as we were already running it on headless chrome?

Comment (BenoitZugmeyer) [reply package.json:23]: I think this would be simpler, yes!

File: test/unit/karma.bs.conf.js [hunk 5]
```diff
@@ -1,11 +1,24 @@
...
   config.set({
     ...karmaBaseConf,
+    files: isExtension ? [...karmaBaseConf.files, 'developer-extension/src/**/*.spec.ts'] : karmaBaseConf.files,
     plugins: [...karmaBaseConf.plugins, 'karma-browserstack-launcher'],
+    preprocessors: isExtension
+      ? {
+          ...karmaBaseConf.preprocessors,
+          'developer-extension/src/**/*.ts': ['webpack', 'sourcemap'],
+        }
+      : karmaBaseConf.preprocessors,
```
Comment (BenoitZugmeyer) [inline test/unit/karma.bs.conf.js:21]: **💬 suggestion:** ‏this is not needed anymore, maybe remove it?

File: rum-events-format
Comment (BenoitZugmeyer) [inline rum-events-format:1]: Changes to the rum-events-format and telemetry event types are unrelated, could you revert them?

**Review by amortemousque — APPROVED** [review]

**Review by nulrich — APPROVED** [review]

**Pull request merged**